src/main/java/services/DatafeedEventsServiceV2.java

```diff
@@ -0,0 +1,89 @@
+package services;
+
+import clients.SymBotClient;
+import model.DatafeedEvent;
+import model.StringId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DatafeedEventsServiceV2 extends AbstractDatafeedEventsService {
+
+    private static final Logger logger = LoggerFactory.getLogger(DatafeedEventsServiceV2.class);
+
+    private final AtomicBoolean started = new AtomicBoolean(false);
+
+    public DatafeedEventsServiceV2(SymBotClient client) {
+        super(client);
+        this.readDatafeed();
+    }
+
+    @Override
+    public void readDatafeed() {
+        try {
+            String datafeedId;
+            List<StringId> datafeedIds = datafeedClient.listDatafeedId();
+            if (datafeedIds.isEmpty()) {
+                datafeedId = this.createDatafeedId();
+            } else {
+                //Each bot should subscribe only one datafeed
+                datafeedId = datafeedIds.get(0).getId();
+            }
+
+            //Read datafeed in loop
+            started.set(true);
+            List<DatafeedEvent> events = null;
+            do {
+                events = datafeedClient.readDatafeed(datafeedId);
```

Comment:
Shouldn't we handle the ackId ? First time : no ackId but we should keep the last one and send it back the next time.

Reply:
I didn't clearly understand about the ackId. As I understand, when we send back the ackId, all events in this datafeed will be deleted. Do we really need this action?

Reply:
Digging from this section : https://perzoinc.atlassian.net/wiki/spaces/AP/pages/392593848/1.+Requirements you might find some more documentation/explanation about `ackId`

Reply:
I believe that this field is related to the AWS/SQS acknowledge mode documented [here](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/getting-started.html#using-client-acknowledge-mode).